blueprints/validator-test/index.js

```diff
@@ -0,0 +1,38 @@
+var path = require('path');
+var existsSync = require('exists-sync');
+var walkSync = require('walk-sync');
+
+module.exports = {
+  description: 'Generates a validator unit test',
+
+  _getFilesPath: function() {
+    var type;
+    if ('ember-cli-mocha' in this.project.addonPackages) {
+      type = 'mocha';
+    } else if ('ember-cli-qunit' in this.project.addonPackages) {
+      type = 'qunit';
+    }
+    return path.join(this.path, type + '-files');
+  },
+
+  // The files and srcPath method can be removed once
+  // https://github.com/ember-cli/ember-cli/pull/4837 is live.
+  // Also, remove the npm dependencies on walk-sync and exists-sync when fixing.
+  files: function() {
```

Comment:
In that case, we would detect in init if we need to use overeidden files and srcPath functions, and if we do we would set them in init. Possibly just via something like this:

    this.files = this._legacyEmberCLIFilesOveride;